intake_erddap/erddap_cat.py

```diff
@@ -26,20 +33,58 @@ class ERDDAPCatalog(Catalog):
     def __init__(
         self,
         server: str,
-        kwargs_search: Optional[dict] = None,
+        kwargs_search: Optional[Dict[str, Union[str, int, float]]] = None,
+        category_search: Optional[List[str]] = None,
         erddap_client: Optional[Type[ERDDAP]] = None,
-        **kwargs
+        **kwargs,
     ):
         """ERDDAPCatalog initialization
 
-        Arguments
-        ---------
-
-            server
+        Parameters
+        ----------
+        server : str
+            ERDDAP server address, for example: "http://erddap.sensors.ioos.us/erddap"
+        kwargs_search : dict, optional
+            Keyword arguments to input to search on the server before making the catalog. Options are:
+            * to search by bounding box: include all of min_lon, max_lon, min_lat, max_lat: (int, float)
+              Longitudes must be between -180 to +180.
+            * to search within a datetime range: include both of min_time, max_time: interpretable
+              datetime string, e.g., "2021-1-1"
+        category_search : list, optional
+            Use this to narrow search by ERDDAP category. The syntax is `[category, key]`, e.g.
+            ["standard_name": "temp"]. `category` is the ERDDAP category for filtering results. Good
+            choices for selecting variables are "standard_name" and "variableName". `key` is the
+            custom_criteria key to narrow the search by, which will be matched to the category results
+            using the custom_criteria that must be set up or input by the user, with `cf-pandas`.
+            Currently only a single key can be matched at a time.
         """
         self._erddap_client = erddap_client or ERDDAP
         self.server = server
+        self.search_url = None
+
+        if kwargs_search is not None:
+            checks = [
+                ["min_lon", "max_lon", "min_lat", "max_lat"],
+                ["min_time", "max_time"],
+            ]
+            for check in checks:
+                if any(key in kwargs_search for key in check) and not all(
+                    key in kwargs_search for key in check
+                ):
+                    raise KeyError(
```

Comment:
The exception should be a `ValueError` because the value passed to the method is wrong, not a key. `KeyError` is raised when `__getitem__` is passed a key that doesn't exist in a mapping to the class.